node-bootstrapper/custom_data_test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCustomData(t *testing.T) {
+	t.Run("ca.crt", func(t *testing.T) {
+		nbc := validNBC()
+		actual := getFile(t, nbc, "/etc/kubernetes/certs/ca.crt", 0600)
+		expected := "test-ca-cert"
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("exec_start.conf", func(t *testing.T) {
+		nbc := validNBC()
+		actual := getFile(t, nbc, "/etc/systemd/system/docker.service.d/exec_start.conf", 0644)
+		nbc.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.DockerBridgeSubnet = "1.1.1.1"
+		expected := `[Service]
```

Comment:
where is the expected custom data coming from ?  

do we need to update this testdata when we change the content of the files ?

Reply:
There is no "source" for the data. Mainly a reflection of what's currently happening.

Yes, it needs to be updated when files are changed.